server/routes/reportDefinition.ts

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import {
+  IRouter,
+  IKibanaResponse,
+  ResponseError,
+} from '../../../../src/core/server';
+import { API_PREFIX } from '../../common';
+import { RequestParams } from '@elastic/elasticsearch';
+import {
+  reportSchema,
+  emailSchema,
+  scheduleSchema,
+  intervalSchema,
+  cronSchema,
+} from '../model';
+import { parseEsErrorResponse } from './utils/helpers';
+import {
+  REPORT_DEF_STATUS,
+  SCHEDULE_TYPE,
+  DELIVERY_CHANNEL,
+} from './utils/constants';
+
+export default function (router: IRouter) {
+  // Create report Definition
+  router.post(
+    {
+      path: `${API_PREFIX}/reportDefinition`,
```

Comment:
what do you think about renaming this to something like `createReportDefinition`? It looks a bit more descriptive to me

Reply:
uri naming shouldn't be descriptive of actions.

Reply:
Gotcha, TIL